File: src/main/java/com/github/weewar/mapviewer/service/WeewarMapLoader.java
package com.github.weewar.mapviewer.service;

import com.github.weewar.mapviewer.exceptions.MapParseException;
import com.github.weewar.mapviewer.model.WeewarMap;

import java.net.URL;
import java.util.List;

public interface WeewarMapLoader {
    WeewarMap load(int mapId) throws MapParseException;

    WeewarMap load(URL mapURL) throws MapParseException;

    List<WeewarMap> loadAll() throws MapParseException;
}

File: src/main/kotlin/com/github/weewar/mapviewer/service/WeewarMapLoader.kt
package com.github.weewar.mapviewer.service

import com.github.weewar.mapviewer.exceptions.MapParseException
import com.github.weewar.mapviewer.model.WeewarMap
import java.net.URL

interface WeewarMapLoader {
    @Throws(MapParseException::class)
    fun load(mapId: Int): WeewarMap

    @Throws(MapParseException::class)
    fun load(mapURL: URL): WeewarMap

    @Throws(MapParseException::class)
    fun loadAll(): List<WeewarMap>
}
